external-crates/move/crates/move-compiler/src/typing/core.rs

```diff
@@ -841,6 +899,23 @@ pub fn make_tvar(_context: &mut Context, loc: Loc) -> Type {
     sp(loc, Type_::Var(TVar::next()))
 }
 
+pub fn make_autoref(context: &mut Context, loc: Loc, ty: Type) -> (RefVar, Type) {
+    use Type_::*;
+    let rv = RefVar::next();
+    let ty = match &ty.value {
+        Type_::Ref(true, inner) => {
+            context.subst.insert_ref_var(rv, RefKind::MutRef);
+            sp(loc, AutoRef(rv, Box::new(*inner.clone())))
+        }
+        Type_::Ref(false, inner) => {
+            context.subst.insert_ref_var(rv, RefKind::ImmRef);
+            sp(loc, AutoRef(rv, Box::new(*inner.clone())))
+        }
+        _ => sp(loc, AutoRef(rv, Box::new(ty.clone()))),
+    };
+    (rv, ty)
+}
```

Comment:
```suggestion
pub fn make_autoref(context: &mut Context, loc: Loc, ty: Type) -> (RefVar, Type) {
    use Type_::*;
    let rv = RefVar::next();
    match &ty.value {
        Ref(true, inner) => {
            context.subst.insert_ref_var(rv, RefKind::MutRef);
            (rv, sp(loc, AutoRef(rv, inner)))
        }
        Ref(false, inner) => {
            context.subst.insert_ref_var(rv, RefKind::ImmRef);
            (rv, sp(loc, AutoRef(rv, inner)))
        }        
        Unit | Param(_) | Apply(_, _) | Fun(_, _) => {
            context.subst.insert_ref_var(rv, RefKind::Value);
            (rv, sp(loc, AutoRef(rv, inner)))
        }
        Anything | UnresolvedError | Var(_) => (rv, sp(loc, AutoRef(rv, Box::new(ty)))),
        AutoRef(existing_rv, _) => (*existing_rv, ty),
    };
    (rv, ty)
}
```

Based on our convo, thoughts on this?

Reply:
Errr fixed this. (if you are looking at an old versio nof the comment I had to edit it because it was wrong